### ISSUE TITLE
[feature/netcore] BadRequest(ModelState) return a bad result called from an OData controller

### ISSUE THREAD

**Issue by LeandroCannizzaro** [OData/WebApi#1376]
In one controller derived from ODataController I try to do a "return a BadRequest(ModelState)".
My Expectation is to have a response at client level as the same code produce in one basic .net core controller as the following json:
```
{
"Name": ["The field Name is required"]
}
```
Instead i have the following response
{"error": {"code":"", "message":""}}
Where the response not is the same that I have in one normal controller and
more important the errors presents in the ModelState not are bounded to the returned response.

*I am using  OData WebApi 7.0.0 Beta 2*

### Reproduce steps
*Create a model with one mandatory field as*
```
    public class TestEntity : Entity
    {
        [Required]
        public string Name { get; set; }
    }

```
*Create a basic ODataController with one method Create*

```
    public class TestApiController : ODataController
    {
        [HttpPost]
        public async Task<IActionResult> Create([FromBody] TestEntity customer)
        {
            if (!ModelState.IsValid)
            {
                //This return not return the validation errors present in the ModelState object.
                return BadRequest(ModelState); 
            }

            //Some code for persist the entity.

            return Created(testEntity);
        }
     }
```
### Expected result
{"Name": ["The field Name is required"]}

### Actual result
{"error""code":"","message":""}}

Thanks in advance for the help.
Leandro

**Comment by robward-ms** [comment]
See #1338 and PR #1343. There was a bug in SerializableErrorExtensions.cs where model state errors were not returned properly. This fix is not in Beta2 but you can try the nightlies to see if that fixes the issue for you.

**Comment by dozer75** [comment]
I had the same issue today, retrieved the latest bits and built, there is a message there, but the message is really poor, doesn't say anything what's wrong when using the Delta type as input parameter.

This is the message I get from the service on ModelState error now:

`{"error":{"code":"","message":" : The input was not valid.\r\n","details":[{"code":"","message":":System.String[]"}]}}`

It isn't much information there WHAT's wrong. When I debug I see that the ModelState is filled by an exception looking like this:

`A null value was found for the property named '<fieldname>', which has the expected type 'Edm.String[Nullable=False]'. The expected type 'Edm.String[Nullable=False]' does not allow null values.`

The ModelState itself is actually not filled correctly either. It should honor how MVC does it otherwise, so I think this logic should be revised.

**Comment by dozer75** [comment]
Added a PR with better error handling, a PR (OData/odata.net#1158) that is requested in the odata.net project will also make this better.

**Comment by xuzhg** [comment]
Please re-open or file new issue if you have further concern. Thanks

**Comment by sergiomoreno** [comment]
> I had the same issue today, retrieved the latest bits and built, there is a message there, but the message is really poor, doesn't say anything what's wrong when using the Delta type as input parameter.
> 
> This is the message I get from the service on ModelState error now:
> 
> `{"error":{"code":"","message":" : The input was not valid.\r\n","details":[{"code":"","message":":System.String[]"}]}}`
> 
> It isn't much information there WHAT's wrong. When I debug I see that the ModelState is filled by an exception looking like this:
> 
> `A null value was found for the property named '<fieldname>', which has the expected type 'Edm.String[Nullable=False]'. The expected type 'Edm.String[Nullable=False]' does not allow null values.`
> 
> The ModelState itself is actually not filled correctly either. It should honor how MVC does it otherwise, so I think this logic should be revised.

Is it this issue solved? It keeps happening to me in version 7.3.
